charts/tfy-karpenter-config/files/al2023-soci-default-provisioner-userdata.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+set -ex
+
+version_lte() {
+  printf '%s\n' "$1" "$2" | sort -C -V
+}
+
+version_lt() {
+  ! version_lte "$2" "$1"
+}
+
+CONTAINERD_VERSION_OUTPUT="$(containerd --version)"
+IFS=" " read -a  CONTAINERD_VERSION_STRING <<< "${CONTAINERD_VERSION_OUTPUT}"
+CONTAINERD_VERSION=${CONTAINERD_VERSION_STRING[2]}
+SANDBOX_IMAGE=$(containerd config dump | grep -oE 'sandbox_image = "(.*)"' | sed 's/sandbox_image = //')
+KUBELET_CONFIG_DIR=/etc/kubernetes/kubelet/config.json.d
+SOCI_KUBELET_CONFIG_PATH=$KUBELET_CONFIG_DIR/99-soci.conf
+CONTAINERD_CONFIG_FILEPATH=/etc/containerd/config.toml
+BACKUP_CONTAINERD_CONFIG_FILEPATH=$CONTAINERD_CONFIG_FILEPATH.bak
+SOCI_RELEASE_VERSION="0.7.0"
+SOCI_CONFIG_DIR="/etc/soci-snapshotter-grpc/"
+SOCI_CONFIG_FILEPATH=$SOCI_CONFIG_DIR/config.toml
+SOCI_SYSTEMD_SERVICE_FILEPATH="/etc/systemd/system/soci-snapshotter.service"
+cp $CONTAINERD_CONFIG_FILEPATH $BACKUP_CONTAINERD_CONFIG_FILEPATH
+
+setup_soci() {( set -ex
+
+yum install fuse -y
+modprobe fuse
+
+curl \
+--silent \
+--show-error \
+--retry 10 \
+--retry-delay 1 \
+-L https://github.com/awslabs/soci-snapshotter/releases/download/v${SOCI_RELEASE_VERSION}/soci-snapshotter-${SOCI_RELEASE_VERSION}-linux-amd64.tar.gz \
+-o soci-snapshotter-$SOCI_RELEASE_VERSION-linux-amd64.tar.gz
```

Comment:
Do these curly braces work fine without quoting the argument? I normally quote the string 😅
 

```suggestion
-L "https://github.com/awslabs/soci-snapshotter/releases/download/v${SOCI_RELEASE_VERSION}/soci-snapshotter-${SOCI_RELEASE_VERSION}-linux-amd64.tar.gz" \
-o "soci-snapshotter-${SOCI_RELEASE_VERSION}-linux-amd64.tar.gz"
```